packages/flutter/lib/src/services/text_boundary.dart

```diff
@@ -180,12 +180,73 @@ class LineBreak extends TextBoundary {
   }
 }
 
+/// A text boundary that uses paragraphs as logical boundaries.
+///
+/// A paragraph is defined as the range between line terminators. If no
+/// line terminator exist then the paragraph boundary is the entire document.
+class ParagraphBoundary extends TextBoundary {
+  /// Creates a [ParagraphBoundary] with the text.
+  const ParagraphBoundary(this._text);
+
+  final String _text;
+
+  @override
+  TextPosition getLeadingTextBoundaryAt(TextPosition position) {
+    return TextPosition(
+      offset: getTextBoundaryAt(position).start,
+    );
+  }
+
+  @override
+  TextPosition getTrailingTextBoundaryAt(TextPosition position) {
+    return TextPosition(
+      offset: getTextBoundaryAt(position).end,
+      affinity: TextAffinity.upstream,
+    );
+  }
+
+  // Returns the [TextRange] representing a paragraph that bounds the given
+  // `textPosition`. The `textPosition` is bounded by either a line terminator
+  // in each direction of the text, or if there is no line terminator in a given
+  // direction then the bound extends to the start/end of the document in that
+  // direction. The returning range includes the line terminator.
```

Comment:
Should these comments use three slashes?  I guess as it is right now, users on the docs site will see the docs for TextBoundary.getTextBoundaryAt, and these comments won't show up anywhere.  Is that intentional?

Reply:
Nit: "returning" => "returned"

Reply:
"the line terminator" => "both line terminators" or "the line terminator in both directions"

Right, it includes both, one at each end?

Reply:
About my first comment, actually I think the docs site will show both the current docs and inherited docs?  See: https://master-api.flutter.dev/flutter/dart-core/List/first.html

So I guess if you did add triple slashes here, the docs site would show both.  Just letting you know in case you choose to do it.

Reply:
I'm okay with triple slashes here, but I noticed the other `TextBoundary` subclasses do not have any additional documentation, so maybe I should just leave it as double slash? Do you have any thoughts @LongCatIsLooong ?